### DISCUSSION TITLE
组件事件中操作另外一个组件的实现方式

### DISCUSSION THREAD

**Discussion by softguy888** [alibaba/formily#2241] [Q&A 2.x] [answered]
请问如何在组件的事件中操作另外一个组件的相关属性，比如在select的onChange事件中，给另外一个组件动态赋值。谢谢！

**Comment by janryWang** [comment, marked as answer]
https://v2.formilyjs.org/zh-CN/guide/advanced/linkages

**Reply by softguy888** [reply]
这个满足不了要求，我要把select中选择的label。写入到另外一个字段，不是写他的值。或者其他固定值

**Reply by janryWang** [reply]
一样的，字段模型里该有的东西都有，你自己取出来，设置到其他字段模型里就行

**Reply by softguy888** [reply]
字段模型里面存的是select的value值，当然enum里面可以获得label，但如果我想在schema的onChange的事件里面直接操作其他的字段，或者获取表单的实例，能否实现。如果是直接写代码肯定是没问题。schema能否做到呢? 谢谢！

**Reply by janryWang** [reply]
该给你说的都说了，其余的自己看文档，总之你说的都能做到

**Comment by hezhengxu2018** [comment]
https://github.com/alibaba/formily/discussions/1337
我觉得你要的是这个，可以在事件中拿到form对象就能操作
